### PR TITLE
Add docs for localization

### PR DESCRIPTION
**Pull request by hishamco**
This includes:

- Add docs for localization abstractions
- Add docs for localization core
- Group extensions into an `Extensions` folder

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Abstractions/Localization/ICalendarManager.cs [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
 {
+    /// <summary>
+    /// Represents a contract for manage calendars.
+    /// </summary>
     public interface ICalendarManager
     {
+        /// <summary>
+        /// Gets the current calendar.
+        /// </summary>
+        /// <returns>The current caldenar name.</returns>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Abstractions/Localization/ICalendarManager.cs]: caldenar

File: src/OrchardCore/OrchardCore.Abstractions/Localization/ICalendarSelector.cs [hunk 2]
```diff
@@ -2,8 +2,15 @@
 
 namespace OrchardCore.Localization
 {
+    /// <summary>
+    /// Represents a contract for selection a calendar.
+    /// </summary>
     public interface ICalendarSelector
     {
+        /// <summary>
+        /// Gets a calednar.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Abstractions/Localization/ICalendarSelector.cs]: calednar

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/HtmlLocalizerExtensions.cs [hunk 3]
```diff
@@ -3,8 +3,19 @@
 
 namespace Microsoft.AspNetCore.Mvc.Localization
 {
+    /// <summary>
+    /// Provides an extension methods for <see cref="IHtmlLocalizer"/>.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/HtmlLocalizerExtensions.cs]: an extension methods => remove `an`

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/HtmlLocalizerExtensions.cs [hunk 4]
```diff
@@ -15,6 +26,13 @@ public static LocalizedHtmlString Plural(this IHtmlLocalizer localizer, int coun
             return localizer[singular, new PluralizationArgument { Count = count, Forms = new[] { singular, plural }, Arguments = arguments }];
         }
 
+        /// <summary>
+        /// Gets the pluralization form.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IHtmlLocalizer"/>.</param>
+        /// <param name="count">the number to be used for selecting the pluralization form.</param>
+        /// <param name="pluralForms">A list of pluralization forms.</param>
+        /// <param name="arguments">The parameters used in the ke.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/HtmlLocalizerExtensions.cs]: in the ke ?

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs [hunk 5]
```diff
@@ -3,8 +3,19 @@
...
     public static class StringLocalizerExtensions
     {
+        /// <summary>
+        /// Gets the pluralization form.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/>.</param>
+        /// <param name="count">the number to be used for selecting the pluralization form.</param>
+        /// <param name="singular">The singular form key.</param>
+        /// <param name="plural">The plural form key.</param>
+        /// <param name="arguments">The parameters used in the ke.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs]: ke => key

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs [hunk 6]
```diff
@@ -15,6 +26,13 @@ public static LocalizedString Plural(this IStringLocalizer localizer, int count,
             return localizer[singular, new PluralizationArgument { Count = count, Forms = new[] { singular, plural }, Arguments = arguments }];
         }
 
+        /// <summary>
+        /// Gets the pluralization form.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/>.</param>
+        /// <param name="count">the number to be used for selecting the pluralization form.</param>
+        /// <param name="pluralForms">A list of pluralization forms.</param>
+        /// <param name="arguments">The parameters used in the ke.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs]: ke => key

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/ViewLocalizerExtensions.cs [hunk 7]
```diff
@@ -3,8 +3,20 @@
 
 namespace Microsoft.AspNetCore.Mvc.Localization
 {
+    /// <summary>
+    /// Provides an extension methods for <see cref="IViewLocalizer"/>.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/ViewLocalizerExtensions.cs]: remove `an`

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/ViewLocalizerExtensions.cs [hunk 8]
```diff
@@ -3,8 +3,20 @@
...
     public static class ViewLocalizerExtensions
     {
+        /// <summary>
+        /// Gets the pluralization form.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IViewLocalizer"/>.</param>
+        /// <param name="count">the number to be used for selecting the pluralization form.</param>
+        /// <param name="singular">The singular form key.</param>
+        /// <param name="plural">The plural form key.</param>
+        /// <param name="arguments">The parameters used in the ke.</param>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/ViewLocalizerExtensions.cs]: please replace all `ke.` by `key`

File: src/OrchardCore/OrchardCore.Localization.Abstractions/IPluralRuleProvider.cs [hunk 9]
```diff
@@ -2,15 +2,24 @@
...
+        /// <remarks>Set the <see cref="Order"/> to a negative value in order to get called before the default implementation, or a 
+        /// positive value to be used as a fallback one.</remarks>
         int Order { get; }
 
+        /// <summary>
+        /// Gets a pluralization rule for the specified culture if there's.
+        /// </summary>
+        /// <param name="culture">The culture.</param>
+        /// <param name="rule">The <see cref="PluralizationRuleDelegate"/>.</param>
+        /// <returns>A boolean value indicates whether the culture rule retrieved or not.</returns>
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/IPluralRuleProvider.cs]: *is* retrieved?

File: src/OrchardCore/OrchardCore.Localization.Abstractions/IPluralRuleProvider.cs [hunk 10]
```diff
@@ -2,15 +2,24 @@
...
     {
+        /// <summary>
+        /// Gets the order for <see cref="IPluralRuleProvider"/> implementation to be executed.
+        /// </summary>
+        /// <remarks>Set the <see cref="Order"/> to a negative value in order to get called before the default implementation, or a 
+        /// positive value to be used as a fallback one.</remarks>
         int Order { get; }
 
+        /// <summary>
+        /// Gets a pluralization rule for the specified culture if there's.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/IPluralRuleProvider.cs]: there `is one`

File: src/OrchardCore/OrchardCore.Localization.Abstractions/IPluralRuleProvider.cs [hunk 11]
```diff
@@ -2,15 +2,24 @@
 
 namespace OrchardCore.Localization
 {
-     /// <summary>
-    /// An implementation of this interface is able to resolve pluralization rules for a specific culture.
-    /// Set the Order to a negative value in order to get called before the default implementation, or a 
-    /// positive value to be used as a fallback one.
+    /// <summary>
+    /// Contract to provider a pluralization rule for cultures.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/IPluralRuleProvider.cs]: to provide

File: src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs [hunk 12]
```diff
@@ -3,17 +3,25 @@
 
 namespace OrchardCore.Localization.PortableObject
 {
+    /// <summary>
+    /// Represents a provider that provides a translations for .po files.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs]: a provider that provides *a* translations

File: src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PortableObjectHtmlLocalizerFactory.cs [hunk 13]
```diff
@@ -4,20 +4,29 @@
 
 namespace OrchardCore.Localization.PortableObject
 {
+    /// <summary>
+    ///
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PortableObjectHtmlLocalizerFactory.cs]: Add description

File: src/OrchardCore/OrchardCore/Localization/NullHtmlLocalizerFactory.cs [hunk 14]
```diff
@@ -7,14 +7,17 @@
 
 namespace OrchardCore.Localization
 {
+    /// Represensts a null <see cref="IHtmlLocalizerFactory"/> which is used by default when the localization module is disabled.
     /// <remarks>
     /// LocalizedHtmlString's arguments will be HTML encoded and not the main string. So the result
     /// should just contained the localized string containing the formatting placeholders {0...} as is.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore/Localization/NullHtmlLocalizerFactory.cs]: contained => contain

File: src/OrchardCore/OrchardCore/Localization/NullStringLocalizerFactory.cs [hunk 15]
```diff
@@ -6,14 +6,17 @@
 
 namespace OrchardCore.Localization
 {
+    /// Represensts a null <see cref="IStringLocalizerFactory"/> which is used by default when the localization module is disabled.
```
Comment (agriffard) [inline src/OrchardCore/OrchardCore/Localization/NullStringLocalizerFactory.cs]: Represensts

**Comment by hishamco** [comment]
Thanks Antonie, I push after the midnight, seems there's a lot of copy paste issue, I will fix them soon

File: src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs (hunk 12 above)
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs]: What's the change here?!!

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs [hunk 16]
```diff
@@ -3,8 +3,19 @@
...
+    /// <summary>
+    /// Provides extension methods for <see cref="IStringLocalizer"/>.
+    /// </summary>
     public static class StringLocalizerExtensions
     {
+        /// <summary>
+        /// Gets the pluralization form.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/>.</param>
+        /// <param name="count">the number to be used for selecting the pluralization form.</param>
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs]: You used a lower case `the`, and you did it in many places in the PR

File: src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs (hunk 12 above)
Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs]: `translation` without `s`

Comment (sebastienros) [reply src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoFilesTranslationsProvider.cs]: Or remove `a`

File: src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs (hunk 16 above)
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Localization.Abstractions/Extensions/StringLocalizerExtensions.cs]: Good catch, it's my bad

I will fix the now

**Comment by hishamco** [comment]
Fixed!!

**Pull request merged**